docs/source/run_model_in_simulation.rst

```diff
@@ -0,0 +1,204 @@
+Run a model in gem5-Aladdin simulation
+======================================
+
+Following the tutorial in :ref:`label_build_python_model`, now we are able to
+create a DL model using the SMAUG Python APIs. In this tutorial, we will
+proceed to introduce the steps of running a model in gem5-Aladdin simulation.
+
+Build gem5-Aladdin
+------------------
+
+First, if you haven't built gem5-Aladdin (cloned in
+:code:`/workspace/gem5-aladdin`), use the following command to build it::
+
+    python2.7 `which scons` build/X86/gem5.opt PROTOCOL=MESI_Two_Level_aladdin -j2
+
+Here, the :code:`MESI_Two_Level_aladdin` is the coherence protocol we will use
+in the gem5 Ruby memory model. Change :code:`-j` parameter to increase the
+number of CPU threads to speed up the build, but keep in mind that you may run
+out of memory before you run out of CPUs. Running out of memory or disk space
+can cause mysterious build failures.
+
+Generate dynamic trace for gem5-Aladdin simulation
+--------------------------------------------------
+
+gem5-Aladdin, which provides the simulation capabilities of estimating
+performance/power/area for pre-RTL kernels via various SoCs, is a trace-driven
+simulator. Thus, in SMAUG, in order to run a model in gem5-Aladdin, we need to
+generate a dynamic trace for the kernels to be simulated as hardware blocks.
+
+First, we need to build the SMAUG tracer, which is an instrumented binary that
+will be executed to generate the dynamic trace::
+
+    make tracer -j4
+
+After running the above command, we will get a binary under :code:`build/bin/`
+named :code:`smaug-instrumented`. Then we will run the model created in
+:ref:`label_build_python_model` using the following command::
+
+    build/bin/smaug-instrumented my_model_topo.pbtxt my_model_params.pb
+
+This generates a trace file named :code:`dynamic_trace_acc0.gz`.
+
+Create gem5-Aladdin configuraion files
+---------------------------------------
+
+gem5-Aladdin also requires two configuration files for running simulations. One
+file specifies SoC-level parameters such as each accelerator's ID, cache or DMA
+configurations, location of the corresponding dynamic trace file, and etc.  An
+example can found be in :code:`experiments/sims/smv/minerva/gem5.cfg`. The
+other file gives the parameters to be used for implementing the accelerator,
+such as parameters for applying loop unrolling and scratchpad partition. An
+example configuration file of implementing the hardware blocks in our `SMV`
+backend can be found in :code:`experiments/sims/smv/smv-accel.cfg`. We defer to
```

Comment:
We should clarify that if the user does not make any changes to the SMV backend, both of these configuration files are sufficient to simulate any model run on SMV. They can change the parameters if they would like, but it may lead to unexpected accelerator performance characteristics, as the array partitioning/loop unrolling/pipelining are all fine tuned to achieve a specific throughput.

Reply:
Added this.